bin/gaia-marionette/gaia-marionette

```diff
@@ -49,47 +52,24 @@ HARNESS_PATHS="$(cat "$BINDIR/../tests/jsmarionette/marionette_tests.txt")"
 
 if [ -z "$TEST_FILES" ]; then
   if [ -z "$APP" ]; then
-    if [ -z "$SKIP_TEST_FILES" ]; then
-      if [ -z "$HARNESS_ONLY" ]; then
-        TEST_FILES=$(find "$BINDIR/.." -path "*/apps/*/test/marionette/*_test.js" \
-                              -or -path "*/dev_apps/*/test/marionette/*_test.js" \
-                              -or -path "*/tv_apps/*/test/marionette/*_test.js" \
-                              -or -path "*/tests/integration/*/*_test.js")
-      else
-        TEST_FILES=""
-      fi
-
-      for wildcard in $HARNESS_PATHS; do
-        HARNESS_FILES=$(find "$BINDIR/../tests/jsmarionette" -path $wildcard)
-        TEST_FILES="$TEST_FILES $HARNESS_FILES"
-      done
```

Comment:
I also simplified the logic here; after all, if `$SKIP_TEST_FILES` is empty, there is no harm to use it, right ? so I removed this part of the if and kept only the second part.

I'm sure we can simplify further more (for example doing the find in a separate function) but I don't want to pack too much stuff here.